src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -205,14 +201,20 @@ private void OnHeadChange(object? sender, BlockReplacementEventArgs e)
             }
         }
 
+        public event EventHandler<BlockReplacementEventArgs>? HeadChanged;
```

Comment:
Having this event here feels super-weird. Can we at least make it somehow work in txpool context?

Reply:
Rename to `TxPoolHeadChanged`?

Reply:
Also used the event